### ISSUE TITLE
Does handles need read from table worker if index lookup doesn't require keep order?

### ISSUE THREAD

**Issue by time-and-fate** [pingcap/tidb#23037]
### tldr

Only read handle column(s) when index lookup requires to keep order => will possibly fail to log missing handles if inconsistency between index and data happened. (just from code logic, haven't verified yet)

Read handle column(s) even when index lookup doesn't require to keep order => sometimes unneeded columns will be read.

Our current implementation is the first case.

---

For the most basic functionality of index lookup, we only need to read handles from the table worker when the index lookup reader requires to keep order.
And this is also what we have implemented on the planner side:
https://github.com/pingcap/tidb/blob/2b977e4141d408a8511d7802427ccf673e3a4c53/planner/core/find_best_task.go#L1032-L1042

However, on the executor side, we have another feature: check inconsistency between data and index. If an inconsistency is found, the corresponding handles from the index and the missing index will be logged:
https://github.com/pingcap/tidb/blob/2b977e4141d408a8511d7802427ccf673e3a4c53/executor/distsql.go#L1128-L1142

To log the missing handles, reading handles from the table worker is necessary. If we always consider the possibility of the inconsistency and want the missing index logged, the handle column(s) from the table worker are always needed no matter whether keep order is required or not, but this will lead to reading unneeded columns sometimes.

Our current implementation is to only read handle column(s) when keep order is required, this might lead to failing to log the missing handles when inconsistency happens.
